app/models/topic.rb

```diff
@@ -32,8 +32,16 @@ def guide_content_ids
     guides.pluck(:content_id).uniq
   end
 
+  def add_section
+    self.topic_sections.build(position: next_position_in_list)
+  end
+
 private
 
+  def next_position_in_list
+    (TopicSection.belonging_to_topic(self.id).maximum(:position) || 0) + 1
+  end
```

Comment:
Very optional but if this was broken into two methods `#highest_position` and `#next_position` (or something like that) it would avoid the `|| 0) + 1` sequence which is a little COMPUTER.